libsplinter/src/admin/service/shared.rs

```diff
@@ -1837,7 +1854,42 @@ mod tests {
         .unwrap();
         let circuit = setup_test_circuit();
 
-        if let Ok(_) = admin_shared.validate_create_circuit(&circuit, b"test_signer_a", "node_a") {
+        let pub_key = (0u8..33).collect::<Vec<_>>();
+        if let Ok(_) = admin_shared.validate_create_circuit(&circuit, &pub_key, "node_a") {
+            panic!("Should have been invalid due to signer not being registered to a node");
+        }
+    }
+
+    #[test]
+    // test that if a circuit is proposed by a signer key is not a valid public key the proposal is
+    // invalid
+    fn test_validate_circuit_signer_key_invalid() {
+        let state = setup_splinter_state();
+        let peer_connector = setup_peer_connector();
+        let orchestrator = setup_orchestrator();
+        let key_registry = StorageKeyRegistry::new("memory".to_string()).unwrap();
+
+        let admin_shared = AdminServiceShared::new(
+            "node_a".into(),
+            orchestrator,
+            peer_connector,
+            Box::new(MockAuthInquisitor),
+            state,
+            Box::new(HashVerifier),
+            Box::new(key_registry),
+            Box::new(AllowAllKeyPermissionManager),
+            "memory",
+        )
+        .unwrap();
+        let circuit = setup_test_circuit();
+
+        let pub_key = (0u8..50).collect::<Vec<_>>();
+        // too short
+        if let Ok(_) = admin_shared.validate_create_circuit(&circuit, &pub_key[0..10], "node_a") {
+            panic!("Should have been invalid due to signer not being registered to a node");
```

Comment:
Can you update the panic messages to say something about the key lengths

Reply:
Updated